test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala

```diff
@@ -247,202 +280,200 @@ class HelpToSaveConnectorSpec extends ControllerSpecWithGuiceApp with HttpSuppor
+            await(result.value).isLeft shouldBe true
+          }
+        }
+      }
+      "the future fails" in {
+        when(GET, getEmailURL, Map.empty, Map.empty).thenReturn(0, "")
+        val result = connector.getEmail()
+        await(result.value).isLeft shouldBe true
+      }
+
     }
 
     "getting account-creation-allowed response" must {
-      behave like testCommon(
-        mockGet(accountCreateAllowedURL),
-        () => connector.isAccountCreationAllowed(),
-        Json.toJson(true),
-        true
-      )
 
       "return a Right if the call comes with HTTP 200 and valid response in the body" in {
-        mockGet(accountCreateAllowedURL)(Some(HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)))
-
+        val response = HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)
+        when(GET, accountCreateAllowedURL, Map.empty, Map.empty).thenReturn(response.status, response.body)
         await(connector.isAccountCreationAllowed().value) shouldBe Right(UserCapResponse())
       }
+
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
+            val response = HttpResponse(status, "")
+            when(GET, accountCreateAllowedURL, Map.empty, Map.empty).thenReturn(response.status, response.body)
+            val result = connector.isAccountCreationAllowed()
+            await(result.value).isLeft shouldBe true
+          }
+        }
+      }
+      "the future fails" in {
+        when(GET, accountCreateAllowedURL, Map.empty, Map.empty).thenReturn(0, "")
+        val result = connector.isAccountCreationAllowed()
+        await(result.value).isLeft shouldBe true
+      }
+
     }
 
     "getting Account" must {
 
       val correlationId = UUID.randomUUID()
-      val url = s"$helpToSaveUrl/help-to-save/$nino/account"
+      val url = s"/help-to-save/$nino/account"
       val queryParameters = Map("correlationId" -> correlationId.toString, "systemId" -> "help-to-save-frontend")
       val account = Account(false, 123.45, 0, 0, 0, LocalDate.parse("1900-01-01"), List(), None, None)
 
-      behave like testCommon(
-        mockGet(url, queryParameters),
-        () => connector.getAccount(nino, correlationId),
-        account,
-        false
-      )
-
       "be able to handle 200 responses with valid Account json" in {
-        mockGet(url, queryParameters)(Some(HttpResponse(200, Json.toJson(account), emptyHeaders)))
+        val response = HttpResponse(200, Json.toJson(account), emptyHeaders)
+        when(GET, url, queryParameters, Map.empty).thenReturn(response.status, response.body)
         await(connector.getAccount(nino, correlationId).value) shouldBe Right(account)
       }
-
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
```

Comment:
same here, can drop the check for != 200

Reply:
Done